### PR TITLE
Created separate production builds for amd and var

### PR DESCRIPTION
**Pull request by sirockin**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks backward-compatibility?    | ✔️
| Tests pass?   | ✔️/❌ - somehow 
| Fixed issues  | #519 (partially, see below)
| Updated README/docs?   | ❌
| Added CHANGELOG entry?   | ❌

The original grunt build produced a single file that could be directly included in html files (by assigning JSONEditor to the window object) or imported into another javascript file using amd.

I don't see any way to achieve this with Webpack - instead it looks like you need to create different builds for the different target formats.

I have followed [this blogpost](https://tech.trivago.com/2015/12/17/export-multiple-javascript-module-formats/) to create a jsoneditor.amd.js build in adition to the other standard ones. However this requires amd users to import the amd version as follows:
```javascript
import { JSONEditor } from '@json-editor/json-editor/dist/jsoneditor.amd.js'
```

I realise this isn't as neat as having both sets of functionality in a single distribution. I'll look into how this can be optimised.
```
:

### CODE REVIEW

File: config/webpack.prod.js [hunk 1]
```diff
@@ -10,61 +10,74 @@ var helpers = require('./helpers');
...
+      /*
+      splitChunks:{
+        chunks:'all'
+      },
+      */
+      minimize:true
+    },
+  
+    plugins: [
+      // new CleanWebpackPlugin({ // Clean all but dev subdirectory before building
```
Comment (schmunk42) [inline config/webpack.prod.js]: Can you remove the commented code

File: config/webpack.prod.js [hunk 2]
```diff
@@ -10,61 +10,74 @@ var helpers = require('./helpers');
...
-    historyApiFallback: true,
-    stats: 'minimal',
-    port:8080
-  }
+  
+    optimization:{
+      // Enabling splitChunks seems to stop the global JSONEditor object being set
+      /*
+      splitChunks:{
+        chunks:'all'
```
Comment (schmunk42) [inline config/webpack.prod.js]: Could this be set explicitly to `none` or something similar

**Comment by pmk65** [comment]
> 
> I realise this isn't as neat as having both sets of functionality in a single distribution. I'll look into how this can be optimised.
> 

I think it's pretty normal to have separate files. The Autocomplete library also distribute several different versions.
https://www.jsdelivr.com/package/npm/@trevoreyre/autocomplete-js?path=dist

**Comment by sirockin** [comment]
@schmunk42 
> btw what does amd and var stand for?

see https://webpack.js.org/configuration/output/#outputlibrarytarget
amd: [Asynchronous Module Definition](https://en.wikipedia.org/wiki/Asynchronous_module_definition)
var: export as global variable (default)

**Comment by schmunk42** [comment]
Thank you!

**Pull request merged**